tests/providers/amazon/aws/operators/test_sagemaker_processing.py

```diff
@@ -0,0 +1,153 @@
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import unittest
+
+import mock
+
+from airflow.exceptions import AirflowException
+from airflow.providers.amazon.aws.hooks.sagemaker import SageMakerHook
+from airflow.providers.amazon.aws.operators.sagemaker_processing import SageMakerProcessingOperator
+
+job_name = 'test-job-name'
+
+create_processing_params = {
+    "AppSpecification": {
+        "ContainerArguments": ["container_arg"],
+        "ContainerEntrypoint": ["container_entrypoint"],
+        "ImageUri": "{{ image_uri }}",
+    },
+    "Environment": {"{{ key }}": "{{ value }}"},
+    "ExperimentConfig": {
+        "ExperimentName": "ExperimentName",
+        "TrialComponentDisplayName": "TrialComponentDisplayName",
+        "TrialName": "TrialName",
+    },
+    "ProcessingInputs": [
+        {
+            "InputName": "AnalyticsInputName",
+            "S3Input": {
+                "LocalPath": "{{ Local Path }}",
+                "S3CompressionType": "None",
+                "S3DataDistributionType": "FullyReplicated",
+                "S3DataType": "S3Prefix",
+                "S3InputMode": "File",
+                "S3Uri": "{{ S3Uri }}",
+            },
+        }
+    ],
+    "ProcessingJobName": job_name,
+    "ProcessingOutputConfig": {
+        "KmsKeyId": "KmsKeyID",
+        "Outputs": [
+            {
+                "OutputName": "AnalyticsOutputName",
+                "S3Output": {
+                    "LocalPath": "{{ Local Path }}",
+                    "S3UploadMode": "EndOfJob",
+                    "S3Uri": "{{ S3Uri }}",
+                },
+            }
+        ],
+    },
+    "ProcessingResources": {
+        "ClusterConfig": {
+            "InstanceCount": 2,
+            "InstanceType": "ml.p2.xlarge",
+            "VolumeSizeInGB": 30,
+            "VolumeKmsKeyId": "{{ kms_key }}",
+        }
+    },
+    "RoleArn": "arn:aws:iam::0122345678910:role/SageMakerPowerUser",
+    "Tags": [{"{{ key }}": "{{ value }}"}],
+}
+
+
+class TestSageMakerProcessingOperator(unittest.TestCase):
+
+    def setUp(self):
+        self.sagemaker = SageMakerProcessingOperator(
+            task_id='test_sagemaker_operator',
+            aws_conn_id='sagemaker_test_id',
+            config=create_processing_params,
+            wait_for_completion=False,
+            check_interval=5
+        )
+
+    def test_parse_config_integers(self):
+        self.sagemaker.parse_config_integers()
+        self.assertEqual(self.sagemaker.config['ProcessingResources']['ClusterConfig']['InstanceCount'],
+                         int(self.sagemaker.config['ProcessingResources']['ClusterConfig']['InstanceCount']))
+        self.assertEqual(self.sagemaker.config['ProcessingResources']['ClusterConfig']['VolumeSizeInGB'],
+                         int(self.sagemaker.config['ProcessingResources']['ClusterConfig']['VolumeSizeInGB']))
```

Comment:
The int-parsing should be tested by `SageMakerBaseOperator` dont you think?

Reply:
What you added was the proper `self.integer_fields`. That's what you want to test.

Reply:
Added the stopping condition back to the unit test.

Reply:
Instead of calling `self.sagemaker.parse_config_integers()` directly, can you test that `self.integer_fields` has the values you expect?

You can use `from parameterized import parameterized` to run a test with different arguments.

Like so..
```suggestion
    @parameterized.expand([
        ([
            ['ProcessingResources', 'ClusterConfig', 'InstanceCount'],
            ['ProcessingResources', 'ClusterConfig', 'VolumeSizeInGB']
        ],),
        ([
            ['ProcessingResources', 'ClusterConfig', 'InstanceCount'],
            ['ProcessingResources', 'ClusterConfig', 'VolumeSizeInGB'],
            ['StoppingCondition', 'MaxRuntimeInSeconds']
        ],),
    ])
    def test_integer_fields_are_set(self, expected_fields):
        assert self.sagemaker.integer_fields == expected_fields
```

But you need to make sure that in one case you pass `StoppingCondition` to config and in the other case don't. You probably would need to change a bit there.